src/wasm2js.h

```diff
@@ -1798,21 +1814,45 @@ Ref Wasm2JSBuilder::processFunctionBody(Module* m,
               break;
             case AndInt32:
               ret = ValueBuilder::makeBinary(left, AND, right);
+              if (curr->type == Type::i32) {
+                return ret;
+              }
```

Comment:
these early returns need a comment, at least on the first - I'm not sure what they do.

Reply:
Yeah, makes sense. Done!